File: VB/WebSite/App_Code/XPObjects.vb
Imports System
Imports DevExpress.Xpo

Public Class Task
	Inherits XPObject

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Public Property AllDay() As Boolean ' Appointment.AllDay
		Get
			Return fAllDay
		End Get
		Set(ByVal value As Boolean)
			SetPropertyValue(NameOf(AllDay), fAllDay, value)
		End Set
	End Property
	Private fAllDay As Boolean


	<Size(SizeAttribute.Unlimited)>
	Public Property Description() As String ' Appointment.Description -  !! To set the Memo field type.
		Get
			Return fDescription
		End Get
		Set(ByVal value As String)
			SetPropertyValue(NameOf(Description), fDescription, value)
		End Set
	End Property
	Private fDescription As String

	Public Property Finish() As Date ' Appointment.End
		Get
			Return fFinish
		End Get
		Set(ByVal value As Date)
			SetPropertyValue(NameOf(Finish), fFinish, value)
		End Set
	End Property
	Private fFinish As Date

	Public Property Label() As Integer ' Appointment.Label
		Get
			Return fLabel
		End Get
		Set(ByVal value As Integer)
			SetPropertyValue(NameOf(Label), fLabel, value)
		End Set
	End Property
	Private fLabel As Integer


	Public Property Location() As String ' Appointment.Location
		Get
			Return fLocation
		End Get
		Set(ByVal value As String)
			SetPropertyValue(NameOf(Location), fLocation, value)
		End Set
	End Property
	Private fLocation As String

	<Size(SizeAttribute.Unlimited)>
	Public Property Recurrence() As String ' Appointment.RecurrenceInfo -  !! To set the Memo field type.
		Get
			Return fRecurrence
		End Get
		Set(ByVal value As String)
			SetPropertyValue(NameOf(Recurrence), fRecurrence, value)
		End Set
	End Property
	Private fRecurrence As String

	<Size(SizeAttribute.Unlimited)>
	Public Property Reminder() As String ' Appointment.ReminderInfo -  !! To set the Memo field type.
		Get
			Return fReminder
		End Get
		Set(ByVal value As String)
			SetPropertyValue(NameOf(Reminder), fReminder, value)
		End Set
	End Property
	Private fReminder As String


	Public Property Created() As Date ' Appointment.Start
		Get
			Return fCreated
		End Get
		Set(ByVal value As Date)
			SetPropertyValue(NameOf(Created), fCreated, value)
		End Set
	End Property
	Private fCreated As Date

	Public Property Status() As Integer ' Appointment.Status
		Get
			Return fStatus
		End Get
		Set(ByVal value As Integer)
			SetPropertyValue(NameOf(Status), fStatus, value)
		End Set
	End Property
	Private fStatus As Integer

	Public Property Subject() As String ' Appointment.Subject
		Get
			Return fSubject
		End Get
		Set(ByVal value As String)
			SetPropertyValue(NameOf(Subject), fSubject, value)
		End Set
	End Property
	Private fSubject As String

	Public Property AppointmentType() As Integer ' Appointment.Type
		Get
			Return fAppointmentType
		End Get
		Set(ByVal value As Integer)
			SetPropertyValue(NameOf(AppointmentType), fAppointmentType, value)
		End Set
	End Property
	Private fAppointmentType As Integer

	Public Property Employee() As Employee ' Appointment.Type
		Get
			Return fEmployee
		End Get
		Set(ByVal value As Employee)
			SetPropertyValue(NameOf(Employee), fEmployee, value)
		End Set
	End Property

	Public fEmployee As Employee
End Class

Public Class Employee
	Inherits XPObject

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	<Size(SizeAttribute.Unlimited)>
	Public Property Name() As String ' Resource.Caption -  !! To set the Memo field type.
		Get
			Return fName
		End Get
		Set(ByVal value As String)
			SetPropertyValue(NameOf(Name), fName, value)
		End Set
	End Property
	Private fName As String
End Class

File: CS/WebSite/App_Code/XPObjects.cs
using System;
using DevExpress.Xpo;

public class Task : XPObject
{
    public Task(Session session) : base(session) { }

    public bool AllDay  // Appointment.AllDay
    {
        get { return fAllDay; }
        set { SetPropertyValue(nameof(AllDay), ref fAllDay, value); }
    }
    bool fAllDay;


    [Size(SizeAttribute.Unlimited)]  // !! To set the Memo field type.    
    public string Description // Appointment.Description
    { 
        get { return fDescription; }
        set { SetPropertyValue(nameof(Description), ref fDescription, value); }
    }
    string fDescription;

    public DateTime Finish // Appointment.End
    {
        get { return fFinish; }
        set { SetPropertyValue(nameof(Finish), ref fFinish, value); }
    }
    DateTime fFinish;

    public int Label // Appointment.Label
    {
        get { return fLabel; }
        set { SetPropertyValue(nameof(Label), ref fLabel, value); }
    }
    int fLabel;


    public string Location // Appointment.Location
    {
        get { return fLocation; }
        set { SetPropertyValue(nameof(Location), ref fLocation, value); }
    }
    string fLocation;          

    [Size(SizeAttribute.Unlimited)]  // !! To set the Memo field type.    
    public string Recurrence // Appointment.RecurrenceInfo
    {
        get { return fRecurrence; }
        set { SetPropertyValue(nameof(Recurrence), ref fRecurrence, value); }
    }
    string fRecurrence;

    [Size(SizeAttribute.Unlimited)]  // !! To set the Memo field type.    
    public string Reminder // Appointment.ReminderInfo
    {
        get { return fReminder; }
        set { SetPropertyValue(nameof(Reminder), ref fReminder, value); }
    }
    string fReminder;

    
    public DateTime Created // Appointment.Start
    {
        get { return fCreated; }
        set { SetPropertyValue(nameof(Created), ref fCreated, value); }
    }
    DateTime fCreated;

    public int Status // Appointment.Status
    {
        get { return fStatus; }
        set { SetPropertyValue(nameof(Status), ref fStatus, value); }
    }
    int fStatus;        

    public string Subject // Appointment.Subject
    {
        get { return fSubject; }
        set { SetPropertyValue(nameof(Subject), ref fSubject, value); }
    }
    string fSubject;    

    public int AppointmentType // Appointment.Type
    {
        get { return fAppointmentType; }
        set { SetPropertyValue(nameof(AppointmentType), ref fAppointmentType, value); }
    }
    int fAppointmentType;

    public Employee Employee // Appointment.Type
    {
        get { return fEmployee; }
        set { SetPropertyValue(nameof(Employee), ref fEmployee, value); }
    }    

    public Employee fEmployee;
}

public class Employee : XPObject
{
    public Employee(Session session) : base(session) { }

    [Size(SizeAttribute.Unlimited)]  // !! To set the Memo field type.
    public string Name // Resource.Caption    
    {
        get { return fName; }
        set { SetPropertyValue(nameof(Name), ref fName, value); }
    }
    string fName;
}

